src/cargo/ops/cargo_compile.rs

```diff
@@ -1526,19 +1527,41 @@ fn find_named_targets<'a>(
     };
 
     if proposals.is_empty() {
-        let targets = packages.iter().flat_map(|pkg| {
-            pkg.targets()
-                .iter()
-                .filter(|target| is_expected_kind(target))
-        });
-        let suggestion = closest_msg(target_name, targets, |t| t.name());
-        anyhow::bail!(
-            "no {} target {} `{}`{}",
+        let targets = packages
+            .iter()
+            .flat_map(|pkg| {
+                pkg.targets()
+                    .iter()
+                    .filter(|target| is_expected_kind(target))
+            })
+            .collect::<Vec<_>>();
+        let suggestion = closest_msg(target_name, targets.iter(), |t| t.name());
+        if !suggestion.is_empty() {
+            anyhow::bail!(
+                "no {} target {} `{}`{}",
+                target_desc,
+                if is_glob { "matches pattern" } else { "named" },
+                target_name,
+                suggestion
+            );
+        }
+        let mut msg = String::new();
+        writeln!(
+            msg,
+            "no {} target {} `{}`.",
             target_desc,
             if is_glob { "matches pattern" } else { "named" },
             target_name,
-            suggestion
-        );
+        )?;
+        if targets.is_empty() {
+            writeln!(msg, "No {} targets available.", target_desc)?;
+            anyhow::bail!(msg);
+        }
```

Comment:
I feel like we should find a way to collapse these messages
- There is a nuance to them that can easily be lost and someone can instead think they are redundant
- The fact that they overlap quite a bit in intention makes it so the signal-to-noise ratio is lower.

Reply:
Make sense. The latter one was deleted.